### PR TITLE
Add Helix definition to guide

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Added Helix definition to guide.

/cc @danmosemsft @karelz @stephentoub

### CODE REVIEW

File: Documentation/project-docs/developer-guide.md [hunk 1]
```diff
@@ -208,7 +208,9 @@ One can build 32- or 64-bit binaries or for any architecture by specifying in th
 
 ### Tests
 
-We use the OSS testing framework [xunit](http://xunit.github.io/) with the [BuildTools test targets](https://github.com/dotnet/buildtools/blob/master/Documentation/test-targets-usage.md).
+We use the OSS testing framework [xunit](http://xunit.github.io/) with the [BuildTools test targets](https://github.com/dotnet/buildtools/blob/master/Documentation/test-targets-usage.md).  
+Tests run inside Microsoft internal-build massive parallel job execution system called **Helix**.
```
Comment (MarcoRossignoli) [inline Documentation/project-docs/developer-guide.md]: @karelz suggestion

File: Documentation/project-docs/developer-guide.md [hunk 2]
```diff
@@ -208,7 +208,9 @@ One can build 32- or 64-bit binaries or for any architecture by specifying in th
 
 ### Tests
 
-We use the OSS testing framework [xunit](http://xunit.github.io/) with the [BuildTools test targets](https://github.com/dotnet/buildtools/blob/master/Documentation/test-targets-usage.md).
+We use the OSS testing framework [xunit](http://xunit.github.io/) with the [BuildTools test targets](https://github.com/dotnet/buildtools/blob/master/Documentation/test-targets-usage.md).  
+Tests run inside Microsoft internal-build massive parallel job execution system called **Helix**.  
+Tests result are show through *Mission Control* reporting site https://mc.dot.net/
```
Comment (danmosemsft) [inline Documentation/project-docs/developer-guide.md]: "results are shown"?

File: Documentation/project-docs/developer-guide.md (hunk 1 above)
Comment (danmosemsft) [reply Documentation/project-docs/developer-guide.md]: "Internal build"-> "internal"

**Comment by jkotas** [comment]
If you just want to have a description of what Helix is somewhere, a good place to add it maybe https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/glossary.md

**Comment by MarcoRossignoli** [comment]
Ok thank's!

File: Documentation/project-docs/developer-guide.md (hunk 2 above)
Comment (MarcoRossignoli) [reply Documentation/project-docs/developer-guide.md]: Yes sorry

File: Documentation/project-docs/developer-guide.md (hunk 1 above)
Comment (MarcoRossignoli) [reply Documentation/project-docs/developer-guide.md]: Ok

File: Documentation/project-docs/glossary.md [hunk 3]
```diff
@@ -139,3 +139,7 @@ The .NET platform currently doesn't contain a cross-platform XAML-based UI stack
 * [WPFLight](https://github.com/ronnycsharp/WPFLight): An OSS project to create WPF on top of XNA/MonoGame.
 * [Perspex](https://github.com/Perspex/Perspex): A cross-platform UI framework based on WPF.
 * [Granular](https://github.com/yuvaltz/Granular): A OSS project to allow WPF applications to run in the browser.
+
+## Engineering system
```
Comment (MarcoRossignoli) [inline Documentation/project-docs/glossary.md]: take "Engineering" term from `eng` folder name. Here maybe we could add other terms `Arcade` etc...

File: Documentation/project-docs/glossary.md [hunk 4]
```diff
@@ -139,3 +139,7 @@ The .NET platform currently doesn't contain a cross-platform XAML-based UI stack
...
+
+* **Helix**. Microsoft internal massive parallel job execution system used to run [xunit](http://xunit.github.io/) tests. Tests result are shown through *Mission Control* reporting site https://mc.dot.net/
```
Comment (jkotas) [inline Documentation/project-docs/glossary.md]: Nit: Helix can run arbitrary workitems, it is not limited to xunit tests. It is a general purpose job processing and result aggregation system running in cloud. It just happens that the workitems that corefx sends to Helix are xunit tests.

File: Documentation/project-docs/glossary.md [hunk 5]
```diff
@@ -139,3 +139,7 @@ The .NET platform currently doesn't contain a cross-platform XAML-based UI stack
 * [WPFLight](https://github.com/ronnycsharp/WPFLight): An OSS project to create WPF on top of XNA/MonoGame.
 * [Perspex](https://github.com/Perspex/Perspex): A cross-platform UI framework based on WPF.
 * [Granular](https://github.com/yuvaltz/Granular): A OSS project to allow WPF applications to run in the browser.
+
+## Engineering system
+
+* **Helix**. It's a massive parallel general purpose job processing and result aggregation system running in cloud. The workitems that corefx sends to Helix are [xunit](https://github.com/xunit) tests. Tests result are shown through *Mission Control* reporting site https://mc.dot.net/
```
Comment (MarcoRossignoli) [inline Documentation/project-docs/glossary.md]: @jkotas feel free to suggest better explanation, I'll copy/paste with no problem.

**Comment by jkotas** [comment]
Thanks!

File: Documentation/project-docs/glossary.md (hunk 5 above)
Comment (safern) [reply Documentation/project-docs/glossary.md]: Might be worth mentioning that to go to the test results in a PR from Azure Devops, you can click on the Send to Helix step in the build and the logs will have the URL.

File: Documentation/project-docs/glossary.md (hunk 4 above)
Comment (safern) [reply Documentation/project-docs/glossary.md]: We actually just changed package tests to run in helix as well which are not xunit tests, but package validation for all of its supported frameworks and RIDs :smile:

File: Documentation/project-docs/glossary.md (hunk 5 above)
Comment (MarcoRossignoli) [reply Documentation/project-docs/glossary.md]: I tried(https://dev.azure.com/dnceng/public/_build/results?buildId=81155&view=logs) ...and links works but test list are empty `No rows currently visible. Please check your filters.` is it expected?

Comment (safern) [reply Documentation/project-docs/glossary.md]: You need to uncheck the "Show failures only" checkbox.

image

Comment (safern) [reply Documentation/project-docs/glossary.md]: Actually having to uncheck that box, is a good thing -> "There where no test failures on that test run" 😆

**Pull request merged**